### PR TITLE
Assignment Client fixes

### PR DESCRIPTION
**Pull request by thoys**
- Sets the SessionUUID before emitting the connectedToDomain signal
     - Could potentially fix multiple issues, but confirmed to fix https://highfidelity.fogbugz.com/f/cases/1768/Agent-AC-fails-to-grab-script-added-from-DS-assignment-run-page
- Fix warning message in assignment clients: `DependencyManager::get(): No instance available for class controller::ScriptingInterface`
- Spelling fixes

## Testing plan:
- Smoke test (make sure to test on your own Sandbox or Standalone Server)
- Test run an assignment script (One comes by default with a clean Sandbox install (ACAudioSearchAndInject.js comes by default, it should make the looping sounds in your Sandbox play, like the fire-crackling)
- Test if the most common defaultScripts still work as expected.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5025b4a5ed51ea47d9e72b5f3d995b58257bf3d4 available at: https://gist.github.com/7cf3e7988ae8454435626b8f64b70762<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8685-5025b4a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8685-5025b4a.dmg

**Comment by marko8904** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5025b4a5ed51ea47d9e72b5f3d995b58257bf3d4 available at: https://gist.github.com/7cf3e7988ae8454435626b8f64b70762<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8685-5025b4a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8685-5025b4a.dmg

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**localhost 'createTestBlocks.js' Result**: **localhost 'createTestBlocks.js'** SUCCESS
**localhost ATP Upload + Edit Result**: **localhost ATP** SUCCESS
**Echo Server Audio**: **Echo Server Audio** SUCCESS
**localhost 'harmonicOscillator.js' Result**: **localhost 'harmonicOscillator.js'** SUCCESS
**localhost 'reverbTest.js' Result**: **localhost 'reverbTest.js'** SUCCESS
**localhost 'playaPerformanceTest.js' Result**: **localhost 'playaPerformanceTest.js'** SUCCESS
**dev-playa 'playaPerformanceTest.js' Result**: **dev-playa 'playaPerformanceTest.js'** SUCCESS
**dev-demo 'playaPerformanceTest.js' Result**: **dev-demo 'playaPerformanceTest.js'** SUCCESS
**QA Pass**: **QA Pass** SUCCESS
**Additional Test Results + Observations**:  Success.  ACAudioSearchAndInject.js worked by default, it make the looping sounds in the Sandbox and the defaultScripts still work as expected.

File: libraries/networking/src/NodeList.cpp [hunk 1]
```diff
@@ -548,13 +553,10 @@ void NodeList::processDomainServerList(QSharedPointer<ReceivedMessage> message)
         // Recieved packet from different domain.
         qWarning() << "IGNORING DomainList packet from" << domainUUID << "while connected to" << _domainHandler.getUUID();
         return;
+    } else {
+        setSessionUUID(newUUID);
```
Comment (Atlante45) [inline libraries/networking/src/NodeList.cpp]: Why not just set those before the `if`?

Comment (thoys) [inline libraries/networking/src/NodeList.cpp]: Because we don't want the sessions ID to change if the DomainList is ignored

Comment (Atlante45) [inline libraries/networking/src/NodeList.cpp]: Gotcha, missed that middle case.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4f7fdfc3451426dad666be48187d34244aad87a4 available at: https://gist.github.com/7cf3e7988ae8454435626b8f64b70762<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8685-4f7fdfc.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8685-4f7fdfc.dmg

**Pull request merged**